### ISSUE TITLE
Support Mistral API

### ISSUE THREAD

**Issue by philippgille** [charmbracelet/mods#174]
Mistral (known for their [7B model](https://mistral.ai/news/announcing-mistral-7b/) and more recently their [Mixture of Experts model](https://mistral.ai/news/mixtral-of-experts/)) have recently started offering an API: <https://docs.mistral.ai/api/>

It would be great if it could be used with mods.

This is similar to the feature request for supporting Ollama (https://github.com/charmbracelet/mods/issues/162) and supporting llamafile (https://github.com/charmbracelet/mods/issues/168).

Maybe one solution could be to use a library that offers an abstraction over all the different LLM APIs already? There's the Go port for LangChain, which supports various LLMs as backend already for example:

- Godoc: https://pkg.go.dev/github.com/tmc/langchaingo@v0.1.0/llms#pkg-overview
- Code: https://github.com/tmc/langchaingo/tree/4200608a750cda0d68015987d6a106da9afd956e/llms

But it doesn't support the Mistral API yet.

**Comment by philippgille** [comment]
Their API seems to be similar to the OpenAI one. I think they use [vLLM](https://github.com/vllm-project/vllm) as the inference/serving engine, which has an OpenAI compatible endpoint (or at least an option to use that, in addition to a non-compatible one?). So I tried with the following mods settings:

```yaml
apis:
  openai:
    // ...
  localai:
    // ...
  mistral:
    base-url: https://api.mistral.ai/v1
    api-key-env: MISTRAL_API_KEY
    models:
      mistral-medium:
        aliases: ["medium"]
        fallback: mistral-small
      mistral-small:
        aliases: ["small"]
        fallback: mistral-tiny
      mistral-tiny:
        aliases: ["tiny"]
// ...
```

And then running with `mods -a mistral -m tiny '...'`, but it printed an OpenAI response for an empty chat message, and ran into an error when saving the conversation ("There was a problem writing 123abc... to the cache. Use  --no-cache  /  NO_CACHE  to disable it.")

It's unclear though what exactly the error is. Did it not find the configured API/model in the settings? Was there an error response from the Mistral API and then it fell back to OpenAI? It doesn't print any details, and mods has no `--verbose` or similar flag to do this.

**Comment by sheldonhull** [comment]
I do this and ollama (which can be mistral) ran fine for me in doing some local scripting..

```yaml
  ollama:
    # LocalAI setup instructions: https://github.com/go-skynet/LocalAI#example-use-gpt4all-j-model
    base-url: http://localhost:11434
    models:
      llama2-13b:
        aliases: ['llama2-13b', 'ollama']
        max-input-chars: 12250

```

**Comment by cloudbridgeuy** [comment]
You can access mistral through `groq`:

```yaml
  groq:
    base-url: https://api.groq.com/openai/v1
    api-key-env: GROQ_API_KEY
    models:
      mixtral-8x7b-32768:
        aliases: ["mixtral", 8x7b]
        max-input-chars: 98000
      llama2-70b-4096:
        aliases: ["llama2"]
        max-input-chars: 12250
```

**Comment by caarlos0** [comment]
using through groq works, yes, ollama should work as well.

pretty much any model with a openai-compatible api should work ✌🏻

**Comment by caarlos0** [comment]
fixed in #305
